src/material/datepicker/datepicker-content.scss

```diff
@@ -2,9 +2,8 @@ $mat-datepicker-calendar-padding: 8px;
 $mat-datepicker-non-touch-calendar-cell-size: 40px;
 $mat-datepicker-non-touch-calendar-width:
     $mat-datepicker-non-touch-calendar-cell-size * 7 + $mat-datepicker-calendar-padding * 2;
-// Based on the natural height of the calendar in a month with 6 rows of dates
-// (largest the calendar will get).
-$mat-datepicker-non-touch-calendar-height: 354px;
+// Height for all content regardless of the number of weeks in a month and the height of the header.
+$mat-datepicker-non-touch-calendar-height: auto;
```

Comment:
I think the reason it was set up with a specific height was so that the calendar panel doesn't jump around as you're navigating between months. Maybe the `auto` height should only apply if there's a custom header?

Reply:
It looks like it. We can also add min-height with this specific value and that resolve it. When there is no custom header, the height will be constant regardless of the number of weeks in a month. When there is a custom header and the weeks will not fit into the container, its height will start to increase.

Reply:
There has been some discussion about this: https://github.com/angular/components/issues/12612